drivers/usb/device/usb_dc_mcux.c

```diff
@@ -74,8 +84,31 @@ K_HEAP_DEFINE_NOCACHE(ep_buf_pool, 1024 * EP_BUF_NUMOF_BLOCKS);
 K_HEAP_DEFINE(ep_buf_pool, 1024 * EP_BUF_NUMOF_BLOCKS);
 #endif
 
+struct usb_ep_ctrl_data {
+	usb_device_callback_message_struct_t transfer_message;
+	struct k_mem_block block;
+	usb_dc_ep_callback callback;
+	uint16_t ep_mps;
+	uint8_t ep_type;
+	uint8_t ep_enabled : 1;
+	uint8_t ep_occupied : 1;
+};
+
+struct usb_dc_state {
+	usb_device_struct_t usb_dev_struct;
+	/* Controller handle */
+	usb_dc_status_callback status_callback;
+	struct usb_ep_ctrl_data *eps;
+	bool attached;
+	uint8_t setupDataStage;
+
+	K_KERNEL_STACK_MEMBER(thread_stack, USBD_MCUX_THREAD_STACK_SIZE);
+
+	struct k_thread thread;
+};
+
 static struct usb_ep_ctrl_data s_ep_ctrl[NUM_OF_EP_MAX];
-static struct usb_device_struct dev_data;
+static struct usb_dc_state dev_data;
```

Comment:
This is still confusing, should probably be renamed to `static struct usb_dc_state dev_state`.

Reply:
Changed.